runtime/kusama/src/xcm_config.rs

```diff
@@ -303,7 +303,9 @@ impl Contains<RuntimeCall> for SafeCallFilter {
 			) |
 			RuntimeCall::XcmPallet(pallet_xcm::Call::limited_reserve_transfer_assets {
 				..
-			}) => true,
+			}) |
+			RuntimeCall::Whitelist(pallet_whitelist::Call::whitelist_call { .. }) |
+			RuntimeCall::Proxy(..) => true,
```

Comment:
The docs of this filter state:

```rust
/// 2. Cannot lead to another call being made;
```
I guess in this case it should be fine since the proxy pallet routes all calls through this filter again?

```rust
/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call parameters.
```
`Proxy::proxy` does accept `call: Box<<T as Config>::RuntimeCall>`, so this is not given.

Reply:
Yes, but we actually still allow `Utility::as_derivative` which is an extrinsic that itself contains a `Call`, in order to allow use cases on both Moonbeam and Acala to still work as intended, so allowing `Proxy::proxy` is also based on the same logic.

In principle however, the 2nd rule may not be necessary, as it's impossible to create an infinitely recursive extrinsic call -- the chain would have thrown an error during decoding as there is a max decoding depth of 10 in the SCALE codec.